gem/lib/mulang/language.rb

```diff
@@ -63,14 +63,24 @@ def initialize(&tool)
     end
 
     def ast(content, **args)
-      @tool.call(content) rescue nil
+      if args[:serialization]
```

Comment:
I'm not sure I get this bit. If serialization is set to true, tool won't be called? what does that mean for extensions like mulang-ruby which use this for parsing?

Reply:
Perhaps it is a bit tricky, but yes, it will be called: `super` calls `ast_analysis`, which calls `base_analysis`, which finally calls `sample`, which polymorphically in this class class `call_tool`.

I think it is ok but just because I am changing implementation semantics here, so that the _main source of truth_  of the ast generation is in `call_tool`, not in `ast` anymore. As a consequence, both methods that require calling the ast generation tool - `sample` and `ast` - delegate on it.

Reply:
However I am not a fan of this implementation, would you implement differently?

Reply:
PS: this was not actually part of this original PR, I should have send as a separate one :rofl:

Reply:
I see!

> However I am not a fan of this implementation, how would you implement it?

If I'm understanding correctly you could just do super in either case then, right? but simply calling tool on second case is faster? I'm fine with that

Reply:
Yes, exactly. The original behavior was to always simply call tool. However, this is not good we when actually want serialization - or maybe some other future flags, like e.g. `force_normalization`, so I added this case.

Reply:
PS:  I have merged this as a separate PR https://github.com/mumuki/mulang/pull/314